contracts/registry/PocketRegistryDelegate.sol

```diff
@@ -14,52 +13,156 @@ contract PocketRegistryDelegate is BaseRegistry {
   address[] public previousDelegates;
 
   // Node state
-  address[] public registeredNodes;
   mapping (address => address) public userNode;
 
   function PocketRegistryDelegate() {
     owner = msg.sender;
   }
 
-  function registerNode() {
+  function createNodeContract(string8[] _supportedTokens, string _url, uint8 _port, bool _isRelayer, bool _isOracle) {
+    require(_supportedTokens.count > 0);
+    require(_url.length > 0);
+    require(_port > 0);
+    require(_isRelayer);
+    require(_isOracle);
 
-    // TODO: Permissions
-    // TODO: Check if address for relay already exists
-    // TODO: Dynamic burn amount
-    // TODO: Check if address is already registered
     tokenAddress.call(bytes4(sha3("burn(uint256,address)")),1,msg.sender);
-    // TODO: instead of using a URL, we need to fetch the node metadata from the node contract
-    register(msg.sender, "URL");
-
-    // TODO: Return the newly created Node
-    // Cannot return values due to delegatecall limitations. See https://ethereum.stackexchange.com/questions/8099/delegatecall-and-function-return-values
-    createNodeContract();
-  }
-
-  function createNodeContract () private {
 
     PocketNode newNode = new PocketNode();
     // TODO: nodeDelegateAddress should only get set once per upgrade
+    newNode.supportedTokens(_supportedTokens);
+    newNode.url(_url);
+    newNode.port(_port);
+    newNode.isRelayer(_isRelayer);
+    newNode.isOracle(_isOracle);
     newNode.changeDelegate(nodeDelegateAddress);
     newNode.setOwner(msg.sender);
     newNode.setTokenAddress(tokenAddress);
     userNode[msg.sender] = newNode;
+
     registeredNodes.push(newNode);
-  }
+    registerNode(newNode.address, newNode.supportedTokens, newNode.url, newNode.port, newNode.isRelayer, newNode.isOracle, registeredNodes.length);
 
-  function getLiveNodes() constant returns (address[]) {
-    return registeredNodes;
   }
 
+  // This is the function that actually insert a record.
+  function registerNode(address key, string8[] _supportedTokens, string _url, uint8 _port, bool _isRelayer, bool _isOracle, uint _index) {
```

Comment:
The index should not be a parameter, because someone can overwrite some else's node. The node index is set as such : `nodeKeys.length - 1`